### PR TITLE
Add meta device support to sparse compressed tensors

### PR DESCRIPTION
**Pull request by pearu**
As in the title.

Unblocks https://github.com/pytorch/pytorch/pull/117907#discussion_r1499251745

Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #120562
* __->__ #120498

cc @alexsamardzic @nikitaved @cpuhrsch @amjames @bhosmer @jcaip

### CODE REVIEW

**Comment by aartbik** [comment]
Thank you so much for doing this so quickly. Much appreciated!

File: c10/core/DispatchKey.h [hunk 1]
```diff
@@ -220,6 +220,7 @@ enum class DispatchKey : uint16_t {
   // TODO: Make SparseCsr a functionality key
```
Comment (ezyang) [inline c10/core/DispatchKey.h:220]: !!!

**Review by ezyang — APPROVED** [review]
thanks, very slick

File: c10/core/DispatchKey.h (hunk 1 above)
Comment (ezyang) [reply c10/core/DispatchKey.h:220]: It's a BE thing. Matters more when we run out of dispatch keys

**Comment by laithsakka** [comment]
does this conflict with https://github.com/pytorch/pytorch/pull/120458

**Pull request closed without merging**